### PR TITLE
PLAT-706 Fix Jempi_API in secure mode

### PR DESCRIPTION
**Pull request by nour-borgi**
Motivation: 

Tested Jempi in AWS secure mode, I wasn't able to login due to cookies issue.
Navigator wasn't saving the CSRF token due to lack of: 
- secure should be true 
- domain shouldn't be the subdomain

In this PR I added two env vars to be overwritten in secure mode.

### CODE REVIEW

File: JeMPI_Apps/JeMPI_API/src/main/resources/application.conf [hunk 1]
```diff
@@ -61,6 +61,19 @@ akka-http-cors {
 
 akka.http.session {
   server-secret = ${JEMPI_SESSION_SECRET}
+  cookie {
+    secure = ${JEMPI_SESSION_SECURE}
+    httpOnly = true
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/src/main/resources/application.conf:66]:
```suggestion
    httpOnly = true
```
I'm not sure about this one.

File: docker/conf/env/create-env-linux-1.sh [hunk 2]
```diff
@@ -38,6 +38,9 @@ export KC_JEMPI_CLIENT_SECRET="Tbe3llP5OJIlqUjz7K1wPp8YDAdCOEMn"
 export KC_JEMPI_ROOT_URL="http://localhost:3000"
 export JEMPI_SESSION_SECRET="c05ll3lesrinf39t7mc5h6un6r0c69lgfno69dsak3vabeqamouq4328cuaekros401ajdpkh60rrt"
 export JEMPI_FILE_IMPORT_MAX_SIZE_BYTE=128000000
+# Deployment related env vars
+export JEMPI_SESSION_SECURE=false
+export DOMAIN_NAME=none
```
Comment (marrouchi) [inline docker/conf/env/create-env-linux-1.sh:43]: shouldn't this be localhost by default ?

File: docker/conf/stack/docker-stack-1.yml [hunk 3]
```diff
@@ -489,6 +489,8 @@ services:
       KC_JEMPI_ROOT_URL: ${KC_JEMPI_ROOT_URL}
       JEMPI_SESSION_SECRET: ${JEMPI_SESSION_SECRET}
       JEMPI_FILE_IMPORT_MAX_SIZE_BYTE: ${JEMPI_FILE_IMPORT_MAX_SIZE_BYTE}
+      JEMPI_SESSION_SECURE: ${JEMPI_SESSION_SECURE}
+      DOMAIN_NAME: ${DOMAIN_NAME}
```
Comment (marrouchi) [inline docker/conf/stack/docker-stack-1.yml:493]: We need maybe to add a prefix "JEMPI_SESSION_"

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: docker/conf/env/create-env-linux-1.sh (hunk 2 above)
Comment (nour-borgi) [reply docker/conf/env/create-env-linux-1.sh:43]: okay cool

File: JeMPI_Apps/JeMPI_API/src/main/resources/application.conf (hunk 1 above)
Comment (nour-borgi) [reply JeMPI_Apps/JeMPI_API/src/main/resources/application.conf:66]: removed it

File: docker/conf/stack/docker-stack-1.yml (hunk 3 above)
Comment (nour-borgi) [reply docker/conf/stack/docker-stack-1.yml:493]: cool

File: JeMPI_Apps/JeMPI_API/src/main/resources/application.conf [hunk 4]
```diff
@@ -61,6 +61,18 @@ akka-http-cors {
...
   server-secret = ${JEMPI_SESSION_SECRET}
+  cookie {
+    secure = ${JEMPI_SESSION_SECURE}
+  }
+  csrf {
+    cookie {
+      name = "XSRF-TOKEN"
+      secure = ${JEMPI_SESSION_SECURE}
+      domain = ${JEMPI_SESSION_DOMAIN_NAME}
+      http-only = false
```
Comment (marrouchi) [inline JeMPI_Apps/JeMPI_API/src/main/resources/application.conf:72]: is this required here as well ?

**Review by marrouchi — CHANGES_REQUESTED** [review]

File: JeMPI_Apps/JeMPI_API/src/main/resources/application.conf (hunk 4 above)
Comment (nour-borgi) [reply JeMPI_Apps/JeMPI_API/src/main/resources/application.conf]: nope, removed it

**Review by MatthewErispe — APPROVED** [review]
@nour-borgi LGTM

**Review by marrouchi — APPROVED** [review]

**Pull request merged**